Mathlib/CategoryTheory/Triangulated/Opposite/Functor.lean

```diff
@@ -0,0 +1,191 @@
+/-
+Copyright (c) 2024 Sophie Morel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sophie Morel
+-/
+import Mathlib.CategoryTheory.Triangulated.Opposite.Pretriangulated
+
+/-!
+# Opposites of functors between pretriangulated categories,
+
+If `F : C ⥤ D` is a functor between pretriangulated categories, we prove that
+`F` is a triangulated functor if and only if `F.op` is a triangulated functor.
+In order to do this, we first show that a `CommShift` structure on `F` naturally
+gives one on `F.op` (for the shifts on `Cᵒᵖ` and `Dᵒᵖ` defined in
+`CategoryTheory.Triangulated.Opposite.Basic`), and we then prove
+that `F.mapTriangle.op` and `F.op.mapTriangle` correspond to each other via the
+equivalences `(Triangle C)ᵒᵖ ≌ Triangle Cᵒᵖ` and `(Triangle D)ᵒᵖ ≌ Triangle Dᵒᵖ`
+given by `CategoryTheory.Pretriangulated.triangleOpEquivalence`.
+
+-/
+
+namespace CategoryTheory
+
+variable {C D : Type*} [Category C] [Category D] [HasShift C ℤ] [HasShift D ℤ] (F : C ⥤ D)
+  [F.CommShift ℤ]
+
+namespace Pretriangulated.Opposite
+
+/-- If `F` commutes with shifts, so does `F.op`, for the shifts chosen on `Cᵒᵖ` in
+`CategoryTheory.Triangulated.Opposite.Basic`.
+-/
+noncomputable scoped instance commShiftOpInt : F.op.CommShift ℤ := by
+  letI F' : OppositeShift C ℤ ⥤ OppositeShift D ℤ := F.op
+  letI : F'.CommShift ℤ := F.commShiftOp ℤ
+  apply F'.commShiftPullback
+
+end Pretriangulated.Opposite
+
+namespace Functor
+
+open Category Limits Pretriangulated Opposite
+
+lemma oppositeCommShiftInt_iso_eq (n m : ℤ) (h : n + m = 0) :
```

Comment:
A better solution would be to start the file as:
```lean
namespace Pretriangulated.Opposite

abbrev commShiftOpIntAux : OppositeShiftAux C ⥤ OppositeShiftAux D := F.op

@[simps! iso]
noncomputable instance instCommShiftOpIntAux : (commShiftOpIntAux F).CommShift ℤ := by
  letI F' : OppositeShift C ℤ ⥤ OppositeShift D ℤ := F.op
  letI : F'.CommShift ℤ := F.commShiftOp ℤ
  apply F'.commShiftPullback

/-- If `F` commutes with shifts, so does `F.op`, for the shifts chosen on `Cᵒᵖ` in
`CategoryTheory.Triangulated.Opposite.Basic`.
-/
noncomputable scoped instance commShiftOpInt : F.op.CommShift ℤ :=
  inferInstanceAs ((commShiftOpIntAux F).CommShift ℤ)

end Pretriangulated.Opposite
```
(which requires defining `OppositeShiftAux` as an `abbrev`, not `private abbrev` in the `Basic` lean.)
Then, we can do
```lean
lemma oppositeCommShiftInt_iso_eq (n m : ℤ) (h : n + m = 0) :
    F.op.commShiftIso n = isoWhiskerRight (shiftFunctorOpIso C n m h) F.op ≪≫
      NatIso.op (F.commShiftIso m).symm ≪≫
        isoWhiskerLeft F.op (shiftFunctorOpIso D n m h).symm := by
  obtain rfl : m = -n := by omega
  exact instCommShiftOpIntAux_iso F n
```